src/Workspaces/Core/Portable/EmbeddedLanguages/DateAndTime/LanguageServices/DateAndTimePatternDetector.cs

```diff
@@ -92,13 +98,7 @@ internal sealed class DateAndTimePatternDetector
             if (!IsMethodArgument(token, syntaxFacts))
                 return false;
 
-            if (!syntaxFacts.IsLiteralExpression(token.Parent))
-                return false;
-
-            if (!syntaxFacts.IsArgument(token.Parent.Parent))
-                return false;
-
-            var argumentList = token.Parent.Parent.Parent;
+            var argumentList = token.Parent!.Parent!.Parent;
```

Comment:
i don't like the `!`s at all.  the benefit of hte previous system was that we didn't need this.

Reply:
Then it is better to remove ```IsMethodArgument```, right?

Reply:
i woudl be fine with that.